tests/check-strings/check-strings

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/python3
+
+# This script checks for strings like paths or class names that are *not* in the source code, but e.g. in
+# translation files, stylesheets or git files.
+# Invalid strings *in the source code* are usually recognized when you compile them, but other strings may
+# be overseen, which is why this script checks strings *outside of the source code*.
+
+
+import re
+import subprocess
+import xml.etree.ElementTree as ElementTree
+from pathlib import Path
+import cssutils
+cssutils.log.enabled = False
+
+
+# global variables
+
+errors = 0
+
+
+# functions
+
+def caption(my_str):
+    print(f'\n# {my_str}\n')
+
+def error(where, my_str):
+    global errors
+    errors += 1
+    print(f'Error: {where}: {my_str}')
+
+# if a string makes a classname, we need to check if that class is in the source
+# however, some of these strings name classes that are not from LMMS, so these can be ignored for such checks:
+def is_our_class(classname: str) -> int:
+    return classname[0] != 'Q'  # Qt classes
+
+
+# prepare some variables
+
+if not Path('.gitmodules').is_file():
+    print('You need to call this script from the LMMS top directory')
+    exit(1)
+
+result = subprocess.run(['git', 'ls-files', '*.[ch]', '*.[ch]pp', '*.ui', ':!tests/*'],
+                        capture_output=True, text=True, check=True)
+files = [Path(f) for f in result.stdout.splitlines()]
+carlabase = 'carlabase' if Path('plugins/carlabase').is_dir() else 'CarlaBase'
+carlapath = f'plugins/{carlabase}/carla/'
+result = subprocess.run(['git',
+                         '--git-dir',   f'{carlapath}/.git',
+                         '--work-tree', f'{carlapath}',
+                         'ls-files', 'resources/ui', 'source/frontend'], capture_output=True, text=True, check=True)
+files.extend([Path(f'{carlapath}/{f}') for f in result.stdout.splitlines()])
+
+classes = set()
+class_pat = re.compile(r'^\s*class(?:\s+LMMS_EXPORT)?\s+([a-zA-Z0-9_]+)', re.MULTILINE)
+class_pat_ui = re.compile(r'<class>([a-zA-Z0-9_]+)</class>')
```

Comment:
Class names can't start with numbers, so `[a-zA-Z0-9_]+` should be `[a-zA-Z_][a-zA-Z0-9_]*` instead.